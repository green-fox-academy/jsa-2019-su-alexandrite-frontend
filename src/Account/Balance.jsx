import React, { useEffect, useState } from 'react';
import {
  View,
  Text,
  TouchableHighlight,
} from 'react-native';
import { useNavigation } from 'react-navigation-hooks';
import { useSelector } from 'react-redux';
import styles from './styles';
import TopUpPopup from './TopUpPopup';

export default function Balance() {
  const { balance } = useSelector((state) => state.user);
  const navigation = useNavigation();
  const [A2WPopupVisible, setA2WPopupVisible] = useState(false);
  const showTopUpPopup = () => setA2WPopupVisible(true);

  useEffect(() => {
    navigation.setParams({
      showTopUpPopup,
    });
  }, []);

  const fn = navigation.getParam('showTopUpPopup');

  return (
    <View>
      <Text style={styles.totalText}>Trading Account Balance</Text>
      <Text style={styles.valueNumber}>{balance}</Text>
      <TouchableHighlight
        style={styles.topUpbutton}
        onPress={fn}
        underlayColor="#0000"
        activeOpacity={0.5}
      >
        <Text style={styles.topUpbuttonText}>TOP-UP</Text>
      </TouchableHighlight>
      <TopUpPopup
        visible={A2WPopupVisible}
        onClose={() => setA2WPopupVisible(false)}
      />
    </View>
  );
}
