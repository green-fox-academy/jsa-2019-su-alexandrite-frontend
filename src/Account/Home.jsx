import React from 'react';
import { View, Text } from 'react-native';

const Home = () => (
  <View>
    <Text>Welcome to our app</Text>
  </View>
);

export default Home;
