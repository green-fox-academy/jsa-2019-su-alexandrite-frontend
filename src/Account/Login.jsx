import React, { useState, useEffect } from 'react';
import {
  TextInput,
  View,
  KeyboardAvoidingView,
  TouchableHighlight,
  Text,
  ActivityIndicator,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigation } from 'react-navigation-hooks';
import { FontAwesome5 } from '@expo/vector-icons';
import PropTypes from 'prop-types';
import { loginUser } from '../redux/account/actionCreator';
import styles from './styles';
import Row from '../common/Row';
import Column from '../common/Column';

const LoginButton = ({ onPress, isLoading }) => (
  <TouchableHighlight style={styles.button} onPress={onPress} underlayColor="#5d70ba" activeOpacity={0.5}>
    {isLoading
      ? <ActivityIndicator size="large" color="white" />
      : (
        <Text style={styles.buttonText}>
          <FontAwesome5 name="arrow-right" size={16} />
        </Text>
      )}
  </TouchableHighlight>
);

const Login = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();
  const { pop } = useNavigation();

  const { accessToken, error, isLoggingIn } = useSelector((state) => state.user);

  useEffect(() => {
    if (accessToken) pop();
  }, [accessToken]);

  const login = () => {
    dispatch(loginUser(username, password));
  };

  return (
    <KeyboardAvoidingView style={styles.container} behavior="padding" enabled>
      <Row style={{ justifyContent: 'center', alignItems: 'center' }}>
        <Text style={styles.loginTitle}>Login</Text>
      </Row>
      <Row style={{ flex: 2 }}>
        <View style={styles.inputContainer}>
          <TextInput
            placeholder="Username"
            label="Username"
            name="username"
            autoCapitalize="none"
            autoCorrect={false}
            style={styles.textInput}
            onChangeText={(text) => setUsername(text)}
            value={username}
          />
          <TextInput
            secureTextEntry
            placeholder="Password"
            label="Password"
            name="password"
            autoCapitalize="none"
            autoCorrect={false}
            style={styles.textInput}
            type="password"
            onChangeText={(text) => setPassword(text)}
            value={password}
          />
          {error
            ? (
              <View style={styles.errorMessageContainer}>
                <Text style={styles.errorMessage}>
                  {error.message}
                </Text>
              </View>
            )
            : null}
          <Row style={{ flex: 0, justifyContent: 'center' }}>
            <Column style={{ flex: 0 }}>
              <LoginButton onPress={login} isLoading={isLoggingIn} />
            </Column>
          </Row>
        </View>
      </Row>
    </KeyboardAvoidingView>
  );
};

LoginButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default Login;
