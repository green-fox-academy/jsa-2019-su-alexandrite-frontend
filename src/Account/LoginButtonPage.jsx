import React from 'react';
import {
  Text, ImageBackground, TouchableHighlight,
} from 'react-native';
import { useNavigation } from 'react-navigation-hooks';
import bg from '../../assets/img/background/login.png';
import Row from '../common/Row';

const LoginButtonPage = () => {
  const { push } = useNavigation();
  return (
    <ImageBackground source={bg} style={{ flex: 1, justifyContent: 'center' }}>
      <Row style={{ justifyContent: 'center', alignItems: 'center' }}>
        <Text style={{
          fontSize: 16,
          fontWeight: 'bold',
          color: '#999',
        }}
        >
          You havenâ€™t logged in
        </Text>
      </Row>
      <Row style={{ justifyContent: 'center', alignItems: 'center' }}>
        <Text style={{
          fontSize: 36,
          fontWeight: 'bold',
          color: '#333',
          textAlign: 'center',
          paddingHorizontal: 60,
        }}
        >
          Login to unlock more features
        </Text>
      </Row>
      <Row flex={0.3} style={{ justifyContent: 'center', alignItems: 'center' }}>
        <TouchableHighlight
          style={{
            paddingHorizontal: 20,
            paddingVertical: 10,
            borderRadius: 20,
            borderWidth: 2,
            borderColor: '#ccc',
          }}
          underlayColor="#eee"
          onPress={() => push('Login')}
        >
          <Text
            style={{
              color: '#999',
              fontWeight: 'bold',
            }}
          >
            TOP-UP
          </Text>
        </TouchableHighlight>
      </Row>
    </ImageBackground>
  );
};

export default LoginButtonPage;
