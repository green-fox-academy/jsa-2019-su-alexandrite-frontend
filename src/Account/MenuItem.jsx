import React from 'react';
import {
  Text,
  TouchableHighlight,
} from 'react-native';
import PropTypes from 'prop-types';
import { FontAwesome5 } from '@expo/vector-icons';

const MenuItem = ({
  name,
  onPress,
  color = '#333',
}) => (
  <TouchableHighlight
    style={{
      flexDirection: 'row',
      flex: 1,
      padding: 15,
      borderBottomColor: '#eee',
      borderBottomWidth: 1,
      justifyContent: 'space-between',
    }}
    underlayColor="#eee"
    onPress={onPress}
  >
    <>
      <Text style={{ flex: 1, color }}>{name}</Text>
      <FontAwesome5 name="chevron-right" />
    </>
  </TouchableHighlight>
);

MenuItem.propTypes = {
  name: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  color: PropTypes.string,
};

MenuItem.defaultProps = {
  color: '#333',
};

export default MenuItem;
