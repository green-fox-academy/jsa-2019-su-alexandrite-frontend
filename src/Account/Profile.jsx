import React from 'react';
import {
  View,
  Text,
} from 'react-native';
import { useSelector } from 'react-redux';
import Row from '../common/Row';
import Card from '../common/Card';

const Profile = () => {
  const {
    username,
  } = useSelector((state) => state.user);

  return (
    <Card style={{ flexDirection: 'row' }}>
      <Row style={{ justifyContent: 'center', marginBottom: 15 }}>
        <View style={{ width: 100, height: 100, borderRadius: 50, backgroundColor: '#ccc' }} />
      </Row>
      <Row style={{ justifyContent: 'center' }}>
        <Text style={{ fontWeight: 'bold', fontSize: 18 }}>{username}</Text>
      </Row>
    </Card>
  );
};

export default Profile;
