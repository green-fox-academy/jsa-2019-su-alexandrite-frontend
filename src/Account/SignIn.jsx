import React, { useState } from 'react';
import {
  View,
  TextInput,
  Button,
  KeyboardAvoidingView,
  Alert,
} from 'react-native';
import { useSelector } from 'react-redux';
import { useNavigation } from 'react-navigation-hooks';
import styles from './styles';

const SignIn = () => {
  const { users } = useSelector((state) => state.users);
  const [name, setName] = useState('');
  const [passWord, setPassWord] = useState('');
  const { push } = useNavigation();

  const signIn = () => {
    if (name === '' || passWord === '') {
      Alert.alert('All fields are requied');
    }

    if (name === 'Aaron' && passWord === '123456') {
      push('Home');
    } else {
      Alert.alert('name or passWord is incorrect');
    }
    // for (let i = 0; i < users.length; i += 1) {
    //   if (name === users[i].userName && passWord === users[i].passWord) {
    //     Alert.alert();
    //   }
    // }
    // Alert.alert();
  };

  return (
    <KeyboardAvoidingView style={styles.loginBox} behavior="padding" enabled>
      <View style={styles.inputContainer}>
        <TextInput
          placeholder="Username"
          label="Username"
          name="username"
          autoCapitalize="none"
          autoCorrect={false}
          style={styles.textInput}
          onChangeText={(text) => setName(text)}
          value={name}
        />
      </View>
      <View style={styles.inputContainer}>
        <TextInput
          secureTextEntry
          placeholder="Password"
          label="Password"
          name="password"
          autoCapitalize="none"
          autoCorrect={false}
          style={styles.textInput}
          type="password"
          onChangeText={(text) => setPassWord(text)}
          value={passWord}
        />
      </View>
      <Button style={styles.button} title="Sign in!" onPress={signIn} />
    </KeyboardAvoidingView>
  );
};

export default SignIn;
