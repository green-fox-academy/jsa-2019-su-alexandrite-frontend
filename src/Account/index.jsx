import React from 'react';
import { View } from 'react-native';
import commonStyles from '../common/styles';
import Balance from './Balance';

export default () => (
  <View style={commonStyles.container}>
    <Balance />
  </View>
);
