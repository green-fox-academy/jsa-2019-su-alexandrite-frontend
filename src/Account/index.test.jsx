/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Account from './index';


describe('<Account />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<Account />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(
      <Account />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
