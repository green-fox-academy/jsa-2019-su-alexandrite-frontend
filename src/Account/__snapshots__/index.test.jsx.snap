// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> Should render and match the snapshot with accessToken 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      Please Login
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          login
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Account /> Should render and match the snapshot without accessToken 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      Please Login
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
